front/app/containers/Admin/formBuilder/components/FormBuilderSettings/LogicSettings/RuleInput.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useState } from 'react';
+
+// components
+import {
+  Box,
+  colors,
+  Icon,
+  IOption,
+  Select,
+  Text,
+} from '@citizenlab/cl2-component-library';
+import Button from 'components/UI/Button';
+
+// intl
+import messages from '../../messages';
+import { FormattedMessage } from 'utils/cl-intl';
+import { Controller, useFormContext } from 'react-hook-form';
+
+type RuleInputProps = {
+  answer: { key: string | undefined; label: string | undefined };
+  name: string;
+  pages:
+    | {
+        value: string | undefined;
+        label: string;
+      }[]
+    | undefined;
+};
+
+type LogicType = { rules: { if: string; goto_page_id: string }[] };
+
+export const RuleInput = ({ pages, name, answer }: RuleInputProps) => {
+  const { setValue, watch, control } = useFormContext();
+  const rules = (watch(name) as LogicType).rules;
+  const initialValue = rules
+    ? rules.filter((rule) => rule.if === answer.key)[0]
```

Comment:
An answer will only ever have 1 rule, so we can use the filter and [0] here. If there is a better way though, just let me know :)